hs_modflow_modelinstance/forms.py

```diff
@@ -0,0 +1,568 @@
+        # will be displayed
+        layout = Layout(
+                        MetadataField('calibratedParameter'),
+                        MetadataField('observationType'),
+                        MetadataField('observationProcessPackage'),
+                        MetadataField('calibrationMethod'),
+        )
+        kwargs['element_name_label'] = 'Model Calibration'
+        super(ModelCalibrationFormHelper, self).__init__(allow_edit, res_short_id, element_id,
+                                                         element_name, layout, *args, **kwargs)
+
+
+class ModelCalibrationForm(ModelForm):
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, **kwargs):
+        super(ModelCalibrationForm, self).__init__(*args, **kwargs)
+        self.helper = ModelCalibrationFormHelper(allow_edit, res_short_id, element_id,
+                                                 element_name='ModelCalibration')
+        self.fields['observationProcessPackage'].choices = observationProcessPackageChoices
+
+    class Meta:
+        model = ModelCalibration
+        fields = ('calibratedParameter',
+                  'observationType',
+                  'observationProcessPackage',
+                  'calibrationMethod',
+                  )
+
+
+class ModelCalibrationValidationForm(forms.Form):
+    calibratedParameter = forms.CharField(max_length=100, required=False)
+    observationType = forms.CharField(max_length=100, required=False)
+    observationProcessPackage = forms.CharField(max_length=100, required=False)
+    calibrationMethod = forms.CharField(max_length=100, required=False)
+
+
+# ModelInput element forms
+class ModelInputFormHelper(BaseFormHelper):
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, element_name=None,
+                 *args, **kwargs):
+        # the order in which the model fields are listed for the FieldSet is the order these fields
+        # will be displayed
+        layout = Layout(
+                        MetadataField('inputType'),
+                        MetadataField('inputSourceName'),
+                        MetadataField('inputSourceURL'),
+        )
+        kwargs['element_name_label'] = 'Model Input'
+        super(ModelInputFormHelper, self).__init__(allow_edit, res_short_id, element_id,
+                                                   element_name, layout, *args, **kwargs)
+
+
+class ModelInputForm(ModelForm):
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, **kwargs):
+        super(ModelInputForm, self).__init__(*args, **kwargs)
+        self.helper = ModelInputFormHelper(allow_edit, res_short_id, element_id,
+                                           element_name='ModelInput')
+        if res_short_id:
+            self.action = "/hsapi/_internal/%s/modelinput/add-metadata/" % res_short_id
+        else:
+            self.action = ""
+
+    @property
+    def form_id(self):

```

Comment:
This is because the resource can have more than one model input so having these properties allows each model input to be updated or deleted individually.